Status/2025Q1/hackathon-202503.adoc: Add report (#494)
freebsd/freebsd-doc: website/content/en/status/report-2025-01-2025-03/hackathon-202503.adoc

Links: +
link:https://wiki.freebsd.org/Hackathon/202503[Hackathon/202503 Wiki Page] URL: link:https://wiki.freebsd.org/Hackathon/202503[]
Links: +
link:https://wiki.freebsd.org/Hackathon[FreeBSD Hackathon Wiki Page] URL: link:https://wiki.freebsd.org/Hackathon[]

Before the link:https://hackmd.io/@AsiaBSDCon/2025lite[AsiaBSDCon-Lite 2025] event, some members of the community gethered and held a hackathon in Tokyo.

Thanks Christoff Visser and Internet Initiative Japan Inc. sponsor the venue.

==== The work done or progressed in the hackathon

===== Sheng-Yi Hung

- ipheth(4): The iPhone tethering use NCM on newer iOS, modifyied ipheth(4) to supporting it. Patch: link:https://reviews.freebsd.org/D49431[]
- Sccahe for FreeBSD base: the FreeBSD base supports ccache to cache the build result. For cross machine build, we need a distributed cache mechanism - that is - scccahe. In Hackathon, the patch for adding sccache support is created: link:https://reviews.freebsd.org/D49417[]

===== Kristof Provost

Wrote a test case for bsnmpd’s snmp_pf module.
This revealed that the BEGEMOT-PF-MIB.txt MIB file could not be parsed by bsnmpwalk, which was also fixed.

> Commits: link:https://cgit.freebsd.org/src/commit/?id=712309a64512c7e4ebf0e10de8a5c59d5a185ae8[712309a64512c7e4ebf0e10de8a5c59d5a185ae8], link:https://cgit.freebsd.org/src/commit/?id=c849f533326026501c28cb2c344b16723862551a[c849f533326026501c28cb2c344b16723862551a] and link:https://cgit.freebsd.org/src/commit/?id=36586800803d24f1137d861bbaf487a6bde16a09[36586800803d24f1137d861bbaf487a6bde16a09”]

Reviewer: ```suggestion
Commits: gitef:712309a64512c7e4ebf0e10de8a5c59d5a185ae8[repository=src], gitref:c849f533326026501c28cb2c344b16723862551a[repository=src] and gitref:36586800803d24f1137d861bbaf487a6bde16a09[repository=src]
```

===== Aymeric Wibo

- Got writing to config space of USB4 routers working and successive reads on AMD USB4 controllers.
- First steps to suspending USB4 routers.
- Put up a bunch of preliminary patches regarding the USB4 stuff.
- Tried passing through USB4 devices to Linux guest to suspend them (did not work).

===== Mark Johnston

- Worked on various syzkaller reports, e.g.: link:https://cgit.freebsd.org/src/commit/?id=fe7fe3b175b626dd1402cd06745b1e3f070c3edd[fe7fe3b175b626dd1402cd06745b1e3f070c3edd]

> - Looked for races in pf after getting some vague bug reports from the OPNsense developers and, with Gleb and Kristof, found and fixed a rare race which could cause a use-after-free: link:https://cgit.freebsd.org/src/commit/?id=8efd2acf07bc0e1c3ea1f7390e0f1cfb7cf6f86c[8efd2acf07bc0e1c3ea1f7390e0f1cfb7cf6f86c]

Reviewer: ```suggestion
- Looked for races in pf after getting some vague bug reports from the OPNsense developers and, with Gleb and Kristof, found and fixed a rare race which could cause a use-after-free: gitref:8efd2acf07bc0e1c3ea1f7390e0f1cfb7cf6f86c[repository=src]
```

===== Philip Paeps

> - Fixed the libtrue website - we now have libtrue.so :-)

Reviewer: ```suggestion
- Fixed the libtrue website -- we now have libtrue.so :-)
```
